### ISSUE TITLE
CMake does not add MAGMA flags in standalone mode

### ISSUE THREAD

**Issue by vqd8a** [kokkos/kokkos-kernels#991]
When building Kokkos Kernels with MAGMA TPL, CMake does not add MAGMA flags to ```CMAKE_CXX_FLAGS```.  The statement ```IF (KOKKOSKERNELS_HAS_TRILINOS)``` in 
https://github.com/kokkos/kokkos-kernels/blob/9c5bbf914eb70bd3d3a6a51eff16615abf7c8d30/cmake/kokkoskernels_tpls.cmake#L453-L463
should be removed to make it work.

**Comment by e10harvey** [comment]
@vqd8a: I am unable to reproduce this when configuring and building kokkos-kernels via:
```bash
cmake -DKokkosKernels_ENABLE_TPL_MAGMA:BOOL=ON -DKokkosKernels_MAGMA_ROOT:PATH=$HOME/local/magma-2.5.4/install ..
make
```

Please provide a reproducer for this bug.

**Comment by vqd8a** [comment]
@e10harvey , Sorry I forgot to provide it
```
[vqdang@white28 kokkoskernels-build-white-magma]$ cmake ~/Kokkos/kokkos-kernels -DCMAKE_CXX_COMPILER=/home/vqdang/Kokkos/kokkos/bin/nvcc_wrapper -DCMAKE_INSTALL_PREFIX=~/Kokkos/kokkoskernels-install-white-magma -DCMAKE_BUILD_TYPE:STRING=RELEASE -DKokkos_DIR=~/Kokkos/kokkos-install-white/lib64/cmake/Kokkos -DKokkosKernels_INST_COMPLEX_DOUBLE:BOOL=ON -DKokkosKernels_ENABLE_TPL_MAGMA:BOOL=ON -DKokkosKernels_MAGMA_ROOT:PATH=$HOME/Packages/magma

[vqdang@white28 kokkoskernels-build-white-magma]$$ make -j install
```
Error message:
```
In file included from /ascldap/users/vqdang/Packages/magma/include/magma_lapack.h:14:0,
                 from /ascldap/users/vqdang/Kokkos/kokkos-kernels/src/impl/tpls/KokkosBlas_tpl_spec.hpp:70,
                 from /ascldap/users/vqdang/Kokkos/kokkos-kernels/src/impl/tpls/KokkosBlas_gesv_tpl_spec_decl.hpp:264,
                 from /ascldap/users/vqdang/Kokkos/kokkos-kernels/src/blas/impl/KokkosBlas_gesv_spec.hpp:162,
                 from /ascldap/users/vqdang/Kokkos/kokkoskernels-build-white-magma/src/impl/generated_specializations_cpp/gesv/Blas_gesv_eti_DOUBLE_LAYOUTLEFT_EXECSPACE_OPENMP_MEMSPACE_HOSTSPACE.cpp:48:
/ascldap/users/vqdang/Packages/magma/include/magma_mangling.h:32:10: error: #error "One of ADD_, NOCHANGE, or UPCASE must be defined to set how Fortran functions are name mangled. For example, in MAGMA, add -DADD_ to CFLAGS, FFLAGS, etc. in make.inc. If using CMake, it defines MAGMA_GLOBAL instead."
         #error "One of ADD_, NOCHANGE, or UPCASE must be defined to set how Fortran functions are name mangled. For example, in MAGMA, add -DADD_ to CFLAGS, FFLAGS, etc. in make.inc. If using CMake, it defines MAGMA_GLOBAL instead."
          ^~~~~
```

**Comment by e10harvey** [comment]
Which magma version and toolchains are you using?

**Comment by vqd8a** [comment]
I am using MAGMA 2.5.1. Let me install MAGMA 2.5.4 to see if I still see these errors.

**Comment by e10harvey** [comment]
@vqd8a I cannot reproduce this on Volta using the cmake command line you've provided above. I configured magma like this:
```
#!/bin/bash
cmake -DCMAKE_CXX_FLAGS=-O3 \
    -DCMAKE_INSTALL_PREFIX=$HOME/local/magma-2.5.4/install \
    -DGPU_TARGET=Volta \
    -DBLA_VENDOR=cuBLAS \
    -DLAPACK_LIBRARIES="$CUDA_ROOT/lib64/liblapack_static.a;$HOME/local/OpenBLAS-0.3.13/install/lib64/libopenblas.a" \

**Comment by vqd8a** [comment]
I did not use cmake when configure MAGMA. I used make.inc instead. But I can try cmake. Thanks @e10harvey

**Comment by e10harvey** [comment]
> I did not use cmake when configure MAGMA. I used make.inc instead. But I can try cmake. Thanks @e10harvey

Ok. That might be the difference. We do not support building without cmake.

**Comment by e10harvey** [comment]
Addressed via #994.
